swap/src/asb/event_loop.rs

```diff
@@ -319,13 +320,40 @@ where
         min_buy: bitcoin::Amount,
         max_buy: bitcoin::Amount,
     ) -> Result<BidQuote> {
+        let monero_decimal = self.monero_wallet.get_balance().await?.as_monero_decimal();
```

Comment:
Nit: The life-time of this variable is unnecessarily long. It is only used from line 334 so there isn't really any reason why this is at the very top!